test/powershell/Language/Parser/LineContinuance.Tests.ps1

```diff
@@ -0,0 +1,247 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe 'Line Continuance' -Tags 'CI' {
+    BeforeAll {
+        function ExecuteCommand {
+            param ([string]$command)
+            [powershell]::Create().AddScript($command).Invoke()
+        }
+    }
+
+    Context 'Lines ending with a backtick that parse and execute without error' {
+        It 'Lines ending with a single backtick' {
+            $script = @'
+'Hello' + `
+    ' world'
+'@
+            ExecuteCommand $script | Should -Be 'Hello world'
+        }
+
+        It 'Lines ending with a single backtick followed by whitespace' {
+            $script = @'
+# The next line ends with trailing whitespace
+'Hello' + `    
+    ' world'
+'@
+            ExecuteCommand $script | Should -Be 'Hello world'
+        }
+
+        It 'Lines ending with a single backtick followed by whitespace and a comment' {
+            $script = @'
+'Hello' + ` # You can place comments after whitespace following backticks
+    ' world'
+'@
+            ExecuteCommand $script | Should -Be 'Hello world'
+        }
+
+        It 'Lines ending with a single backtick followed by a comment line and then the continued line' {
+            $script = @'
+'Hello' + `
+# You can place comments in the middle of a continued line
+    ' world'
+'@
+            ExecuteCommand $script | Should -Be 'Hello world'
+        }
+    }
+
+    Context 'Lines ending with a backtick that do not parse' {
+        It 'Lines ending with a single backtick followed immediately by a comment' {
+            $script = @'
+'Hello' + `# This is not a valid comment
+    ' world'
+'@
+            $err = { ExecuteCommand $script } | Should -Throw -ErrorId 'ParseException' -PassThru
+            $err.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should -BeExactly 'ExpectedValueExpression'
+        }
+
+        It 'Lines ending with two backticks' {
+            $script = @'
+'Hello' + ``
+    ' world'
+'@
+            $err = { ExecuteCommand $script } | Should -Throw -ErrorId 'ParseException' -PassThru
+            $err.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should -BeExactly 'ExpectedValueExpression'
+        }
+    }
+
+    Context 'Lines ending with a pipe that parse and execute without error' {
+        It 'Lines ending with a pipe' {
+            $script = @'
+'Hello' |
+    ForEach-Object {"$_ world"}
+'@
+            ExecuteCommand $script | Should -Be 'Hello world'
+        }
+
+        It 'Lines ending with a pipe followed by whitespace' {
+            $script = @'
+# The next line ends with trailing whitespace
+'Hello' |     
```

Comment:
End spaces can be auto removed in editors. We should protect. Please use operators to construct the test string.

Reply:
Good point, done.

Reply:
I don't see your update.

Reply:
See https://github.com/PowerShell/PowerShell/pull/8938/commits/ad12433f43edee02090efb1b4ef17ac97c61e9fd#diff-6d2b8b1bd72c210d3be378d71e27a146.

Reply:
It is fix for line 25 - line 82 is not fixed.

Reply:
Sorry, missed that one. Commit coming soon.